metrics/influxdb.go

```diff
@@ -0,0 +1,94 @@
+package metrics
+
+import (
+	"fmt"
+	"sync"
+
+	gomet "github.com/rcrowley/go-metrics"
+	"github.com/vrischmann/go-metrics-influxdb"
+)
+
+// Influx :
+type Influx struct {
+	Counter                                map[string]gomet.Counter
+	Histogram                              map[string]gomet.Histogram
+	threadLock, counterLock, histogramLock *sync.Mutex
+	running                                *bool
+}
+
+// SendMetricsNow : implements Metrics interface
+func (influx Influx) SendMetricsNow() {
+	influxdb.SendMetricsNow()
+}
+
+// New :
+func (influx Influx) New() Influx {
+	influx.threadLock, influx.counterLock, influx.histogramLock = new(sync.Mutex), new(sync.Mutex), new(sync.Mutex)
+	counter := make(map[string]gomet.Counter)
+	histrogram := make(map[string]gomet.Histogram)
+	counter[Requests] = gomet.NewCounter()
+	gomet.Register(Requests, counter[Requests])
+	counter[Successes] = gomet.NewCounter()
+	gomet.Register(Successes, counter[Successes])
+	sample := gomet.NewExpDecaySample(1028, 0.015)
+	histrogram[LatencyHistogram] = gomet.NewHistogram(sample)
+	gomet.Register(LatencyHistogram, histrogram[LatencyHistogram])
+	influx.Counter = counter
+	influx.Histogram = histrogram
+	status := false
+	influx.running = &status
+	return influx
+}
+
+// Monitor : implement Metrics
+func (influx Influx) Monitor(opts *ServerOpts) {
+	if *influx.running {
+		fmt.Println("monitor has already running")
+		return
+	}
+	influx.threadLock.Lock()
+	*influx.running = true
+	chRelease := make(chan bool, 1)
+
+	go func() {
+		influxdb.InfluxDB(
+			gomet.DefaultRegistry,
+			opts.WriteInterval,
+			opts.Host,
+			opts.Database,
+			opts.Username,
+			opts.Password)
+
+		defer func() {
+			if r := recover(); r != nil {
```

Comment:
it will regain control after panic occur, it just work like exception handling. 
see [Defer, Panic and Recover](https://blog.golang.org/defer-panic-and-recover)